src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -409,6 +416,34 @@ func (q *qemuAmd64) setupSEVGuestPreAttestation(ctx context.Context, config sev.
 	return attestationId, nil
 }
 
+func getCPUSig(cpuModel string) sev.VCPUSig {
+	// This is for the special case for SNP (see cpuModel()).
+	if cpuModel == "EPYC-v4" {
+		return sev.SigEpycV4
+	}
+	return sev.NewVCPUSig(cpuid.DisplayFamily, cpuid.DisplayModel, cpuid.SteppingId)
+}
+
+func calculateGuestLaunchDigest(config sev.GuestPreAttestationConfig, numVCPUs int, cpuModel string) ([sha256.Size]byte, error) {
+	if config.Policy&sevPolicyBitSevEs != 0 {
+		// SEV-ES guest
+		return sev.CalculateSEVESLaunchDigest(
+			numVCPUs,
+			getCPUSig(cpuModel),
```

Comment:
Would it be nicer to avoid exposing the special AMD CPU sig here? We could move the conversion to the AMD Sig inside `sev.CalculateSEVESLaunchDigest` and just use the more generic cpuModel here? I don't have a strong preference.

Reply:
Your suggestion would be nicer from the kata runtime point-of-view, but if I try to keep `pkg/sev` as a "standalone Go pkg for generic SEV-related stuff like calculating launch digests", then the current function interface make more sense. I wanted to keep the package more standalone-y, and leave the qemu-related stuff in this file (`virtcontainers/qemu_amd64.go`).